hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleServiceImpl.java

```diff
@@ -32,4 +43,64 @@ public void registerSchemas(@NonNull final SchemaRegistry registry) {
         registry.register(new V0490ScheduleSchema());
         registry.register(new V0570ScheduleSchema());
     }
+
+    @Override
+    public Iterator<ExecutableTxn> iterTxnsForInterval(
+            final Instant start, final Instant end, final Supplier<StoreFactory> cleanupStoreFactory) {
+        final var store = cleanupStoreFactory.get().writableStore(WritableScheduleStoreImpl.class);
+        // Filter transactions that are not executed/deleted and have all the required keys
+        final var executableTxns = store.getByExpirationBetween(start.getEpochSecond(), end.getEpochSecond()).stream()
+                .filter(schedule -> !schedule.executed() && !schedule.deleted())
+                .toList();
+
+        // Return a custom iterator that supports the remove() method
+        return new Iterator<>() {
+            private int currentIndex = -1; // To keep track of the current element
+            private final List<Schedule> transactions = new ArrayList<>(executableTxns);
```

Comment:
I don't see a reason to copy the list to a mutable collection; the semantics of this `Iterator.remove()` implementation is simply:
 - If `remove()` is called, the last-returned `Schedule` is **purged from state using a store created by the `Supplier<StoreFactory>`.

It's not important to remove the `Schedule` from a backing list.

Reply:
Changed.